### PR TITLE
EXLM 608 ADLS Block

### PR DESCRIPTION
**Pull request by devisree29**
ADLS Card Block

Jira ID:https://jira.corp.adobe.com/browse/EXLM-608

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.page/en/adls-cards-page
- After: https://feature-exlm-608-adls--exlm--adobe-experience-league.hlx.page/en/adls-cards-page
image

### CODE REVIEW

File: blocks/adls-cards/adls-cards.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  // Appending header div to the block
+  block.appendChild(headerDiv);
+  const contentDiv = document.createElement('div');
+  contentDiv.classList.add('browse-cards-block-content');
+  try {
+    await loadIms();
+  } catch {
+    // eslint-disable-next-line no-console
+    console.warn('Adobe IMS not available.');
+  }
```
Comment (noormd08) [inline blocks/adls-cards/adls-cards.js:46]: Do we need Adobe IMS for ADLS Block?

**Review by ruhisingh1 — DISMISSED** [review]

File: scripts/browse-card/browse-cards-delegate.js [hunk 2]
```diff
@@ -127,9 +127,14 @@ const BrowseCardsDelegate = (() => {
     const urlSearchParams = new URLSearchParams();
     urlSearchParams.append('trainingMethod', 'Live Instructor Courses');
     urlSearchParams.append('pageIndex', '1');
-    urlSearchParams.append('sort', 'recommended');
     urlSearchParams.append('learningType', 'catalog');
-    return urlSearchParams;
+    if (param.solutions !== '') {
```
Comment (noormd08) [inline scripts/browse-card/browse-cards-delegate.js:131]: Just go with **if (param.solutions)** as this has to go if we pass empty string.

File: scripts/browse-card/browse-cards-delegate.js [hunk 3]
```diff
@@ -127,9 +127,14 @@ const BrowseCardsDelegate = (() => {
...
+      urlSearchParams.append('products', param.solutions);
+    }
+    if (param.roles !== '') {
+      urlSearchParams.append('roles', param.roles);
```
Comment (noormd08) [inline scripts/browse-card/browse-cards-delegate.js:135]: Same as above

File: blocks/adls-cards/adls-cards.js (hunk 1 above)
Comment (devisree29) [reply blocks/adls-cards/adls-cards.js]: Removed the Code related to Adobe IMS

File: scripts/browse-card/browse-cards-delegate.js (hunk 2 above)
Comment (devisree29) [reply scripts/browse-card/browse-cards-delegate.js]: Made the required Changes

File: scripts/browse-card/browse-cards-delegate.js (hunk 3 above)
Comment (devisree29) [reply scripts/browse-card/browse-cards-delegate.js:135]: Made the required Changes

**Review by noormd08 — APPROVED** [review]

**Pull request merged**